src/sql/src/plan/statement/ddl.rs

```diff
@@ -1151,9 +1140,29 @@ pub fn plan_create_source(
         seen: _,
     } = CreateSourceOptionExtracted::try_from(with_options.clone())?;
 
-    let (key_desc, value_desc) = encoding.desc()?;
+    let (key_desc, value_desc) = match &encoding {
+        Some(encoding) => {
+            let (key_desc, value_desc) = encoding.desc()?;
+
+            // TODO(petrosagg): get rid of this logic
```

Comment:
Yes, sorry for the cryptic comment. The context is that there is a much bigger PR ([this one](https://github.com/MaterializeInc/materialize/pull/24626)) from which this and this are smaller parts of it that I split off to make reviewing easier. This logic that I commented as having to get rid of here is already gone in the big PR and so the lifetime of this comment (and logic) is basically a week.

As for the reason why this logic is undesirable that is because this is the planner making a decision based on the specific source type selected but it should instead be expressing the true logic based on the relation description produced by the source.

Reply:
Added a much longer comment explaining what I said above